File: fundamentals/Part 1 - Data Preprocessing/section2/data_preprocessing_template.py
# Data Preprocessing Template

import pandas as pd

from sklearn.cross_validation import train_test_split

# Importing the dataset
dataset = pd.read_csv('Data.csv')
X = dataset.iloc[:, :-1].values
y = dataset.iloc[:, 3].values

# taking care of missing data removed

# Encoding (removed)

#Splitting the dataset into the Training set and Test set
#20% chosen as our test observations
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.2, random_state = 0)

# Feature scaling
'''Scaling data for data that are on different scales to avoid calculations discrepencies
 due to the euclidian distance which is based on squares'''
 #comment out cuse not always needed

'''from sklearn.preprocessing import StandardScaler
scale_X = StandardScaler()
X_train = scale_X.fit_transform(X_train)
X_test = scale_X.transform(X_test)'''

File: fundamentals/Part 1 - Data Preprocessing/section2/data_preprocessing_template.R
# Data Preprocessing Template
library(caTools)
# Importing the dataset
dataset = read.csv('Data.csv')


#Splitting the dataset into the Training set and Test set using caTools

set.seed(123)
split = sample.split(dataset$Purchased, SplitRatio = 0.8)
training_set = subset(dataset, split == TRUE)
test_set = subset(dataset, split == FALSE)

# Feature scaling
#only scaling col 2 and 3 (1 and 4 are factors not numeric)
# training_set[, 2:3] = scale(training_set[, 2:3] )
# test_set[, 2:3]  = scale(training_set[, 2:3] )

